### PR TITLE
Add Path2D/3D debug options

### PR DESCRIPTION
**Pull request by smix8**
Adds Path2D/3D debug visuals and options for Editor Debug menu, engine parameters, ProjectSettings and runtime visuals for 3d (2d already had runtime visuals).

Path curves used the same debug options as navigation. A fixed color with navigation debug was sometimes a visual problem as it would melt with the navmesh debug but also building Godot without the navigation module was not possible. The path debug required certain properties from the navigation and with the rework of the navigation debug on the horizon a split was necessary.

path_debug_options

path_debug_options2

I am not happy with the thin line visuals of the 3D PRIMITIVE_LINES debug but hesitant to build a full triangle mesh as paths can include so many points. Would love to see a fix for the primitive line width rendering some day.

### CODE REVIEW

**Comment by Calinou** [comment]
> I am not happy with the thin line visuals of the 3D PRIMITIVE_LINES debug but hesitant to build a full triangle mesh as paths can include so many points. Would love to see a fix for the primitive line width rendering some day.

Unlike OpenGL, Vulkan expects you to reimplement line width using triangle strips if I'm not mistaken. Line width under OpenGL was often limited to specific graphics drivers (e.g. AMD never supported line widths greater than 1).

**Comment by smix8** [comment]
A build-in mesh options for lines would be very welcome. Perfect if it has an added outline option for contrast as it is a constant struggle to spot the thin line primitives on certain backgrounds. The selections boxes, while a little better, have the same issues that in certain scenes they melt with the background and might as well not exist.

**Comment by Calinou** [comment]
> Perfect if it has an added outline option for contrast as it is a constant struggle to spot the thin line primitives on certain backgrounds. The selections boxes, while a little better, have the same issues that in certain scenes they melt with the background and might as well not exist.

I suppose one of the selection box lines could be made darker/lighter to avoid this.

File: scene/main/scene_tree.h [hunk 1]
```diff
@@ -146,9 +147,12 @@ class SceneTree : public MainLoop {
 
 	Color debug_collisions_color;
 	Color debug_collision_contact_color;
+	Color debug_paths_color;
+	float debug_paths_width;
```
Comment (akien-mga) [inline scene/main/scene_tree.h:151]:
```suggestion
	float debug_paths_width = 1.0f;
```

**Review by akien-mga — APPROVED** [review]

**Pull request merged**